deltachat_derive/src/lib.rs

```diff
@@ -8,36 +8,33 @@ use quote::quote;
 // data.  If this assumption is violated, compiler error will point to
 // generated code, which is not very user-friendly.
 
-#[proc_macro_derive(ToSql)]
-pub fn to_sql_derive(input: TokenStream) -> TokenStream {
+#[proc_macro_derive(Sqlx)]
+pub fn sqlx_derive(input: TokenStream) -> TokenStream {
     let ast: syn::DeriveInput = syn::parse(input).unwrap();
     let name = &ast.ident;
 
     let gen = quote! {
-        impl rusqlite::types::ToSql for #name {
-            fn to_sql(&self) -> rusqlite::Result<rusqlite::types::ToSqlOutput> {
-                let num = *self as i64;
-                let value = rusqlite::types::Value::Integer(num);
-                let output = rusqlite::types::ToSqlOutput::Owned(value);
-                std::result::Result::Ok(output)
+        impl sqlx::encode::Encode<sqlx::sqlite::Sqlite> for #name {
+            fn encode(&self, buf: &mut Vec<sqlx::sqlite::SqliteArgumentValue>) {
+                num_traits::ToPrimitive::to_i64(self).expect("invalid type").encode(buf)
             }
         }
-    };
-    gen.into()
-}
 
-#[proc_macro_derive(FromSql)]
-pub fn from_sql_derive(input: TokenStream) -> TokenStream {
-    let ast: syn::DeriveInput = syn::parse(input).unwrap();
-    let name = &ast.ident;
 
-    let gen = quote! {
-        impl rusqlite::types::FromSql for #name {
-            fn column_result(col: rusqlite::types::ValueRef) -> rusqlite::types::FromSqlResult<Self> {
-                let inner = rusqlite::types::FromSql::column_result(col)?;
-                Ok(num_traits::FromPrimitive::from_i64(inner).unwrap_or_default())
+        impl<'de> sqlx::decode::Decode<'de, sqlx::sqlite::Sqlite> for #name {
+            fn decode(value: sqlx::sqlite::SqliteValue<'de>) -> sqlx::Result<Self> {
+                let raw: i64 = sqlx::decode::Decode::decode(value)?;
+
+                Ok(num_traits::FromPrimitive::from_i64(raw).unwrap_or_default())
```

Comment:
Would it not make more sense to return an error instead of default, i.e. 0?

Reply:
We rely on this logic quite heavily atm, this is trying to match was there before.

Reply:
i got here by looking how `MsgId` was replaced, and that used to error in it's custom tosql/fromsql impl.  Now it doesn't.  So as a blanket statement that's not entirely true.

Reply:
true, I guess I missed that difference